opm/material/fluidmatrixinteractions/PiecewiseLinearTwoPhaseMaterialParams.hpp

```diff
@@ -242,4 +245,24 @@ class PiecewiseLinearTwoPhaseMaterialParams : public EnsureFinalized
 };
 } // namespace Opm
 
+namespace Opm::cuistl{
+
+template <class TraitsT, class ContainerType, class ViewType>
+PiecewiseLinearTwoPhaseMaterialParams<TraitsT, ViewType> make_view(const PiecewiseLinearTwoPhaseMaterialParams<TraitsT, ContainerType>& params) {
+    ViewType SwPcwnSamples = make_view<const float, float>(params.SwPcwnSamples());
```

Comment:
Should this be float unconditionally, and not depend on the underlying scalar type requested in the ViewType?

Reply:
Good point, It should depend on the viewType which should store something that matches the containerType

Reply:
I see a static_assert in your future...

Reply:
I have changed the type and added the static assert now